tests/unit/test_openstack.py

```diff
@@ -594,3 +601,44 @@ async def test_network_picking_no_spread(self, x):
         await self.translate_network_types_test_core(
             x, "mrack_no_spread.conf", "no", 95
         )
+
+    @pytest.mark.asyncio
+    async def test_openstack_gather_responses(self):
+        error_flag = True
+
+        async def coro1():
+            await asyncio.sleep(0.1)
+            return "result1"
+
+        async def coro2():
+            nonlocal error_flag
+            await asyncio.sleep(0.2)
+            if error_flag:
+                error_flag = False
+                raise ServerError("Oops!", 503)
+            else:
+                return "result2"
+
+        async def coro3():
+            await asyncio.sleep(0.3)
+            raise ServerError("Oops!", 503)
+
+        provider = OpenStackProvider()
+
+        # Test successful outcome
+        result = await provider._openstack_gather_responses([coro1(), [], {}])
+        assert result[0] == "result1"
+
+        # Test that we get successful outcome after getting a ServerError
+        error_flag = True
+        result = await provider._openstack_gather_responses(
+            [coro1(), [], {}], [coro2(), [], {}]
```

Comment:
I think you forgot brackets here
```suggestion
            [coro1, [], {}], [coro2, [], {}]
```

Reply:
thanks, fixed